lib/private/AllConfig.php

```diff
@@ -491,6 +491,45 @@ public function getUsersForUserValue($appName, $key, $value) {
 		return $userIDs;
 	}
 
+	/**
+	 * Gets the list of user ids sorted by lastLogin, from most recent to least recent
+	 *
+	 * @param int|null $limit how many users to fetch
+	 * @param int $offset from which offset to fetch
+	 * @param string $search search users based on search params
+	 * @return array of user IDs
+	 */
+	public function getLastLoggedInUsers(?int $limit = null, int $offset = 0, string $search = ''): array {
+		// TODO - FIXME
+		$this->fixDIInit();
+
+		$query = $this->connection->getQueryBuilder();
+
+		$query->select('uid', 'displayname')
```

Comment:
```suggestion
		$query->selectDistinct('uid')
```
You do not use displayname so no need to select it.
And if you use selectDistinct you can avoid using uid as key in the foreach loop below and can return a list which is better than an associative array.
You can even return $result->fetchAll in column mode.

Reply:
used selectDistinct and fetchAll